Sources/GRPC/GRPCServerPipelineConfigurator.swift

```diff
@@ -103,18 +103,22 @@ final class GRPCServerPipelineConfigurator: ChannelInboundHandler, RemovableChan
       channel: channel,
       targetWindowSize: self.configuration.httpTargetWindowSize
     ) { stream in
-      // TODO: use sync options when NIO HTTP/2 support for them is released
-      // https://github.com/apple/swift-nio-http2/pull/283
-      stream.getOption(HTTP2StreamChannelOptions.streamID).map { streamID -> Logger in
-        logger[metadataKey: MetadataKey.h2StreamID] = "\(streamID)"
-        return logger
-      }.recover { _ in
-        logger[metadataKey: MetadataKey.h2StreamID] = "<unknown>"
-        return logger
-      }.flatMap { logger in
+      // Sync options were added to the HTTP/2 stream channel in 1.17.0 (we require at least this)
+      // so this shouldn't be `nil`, but it's not a problem if it is.
+      let http2StreamID = try? stream.syncOptions?.getOption(HTTP2StreamChannelOptions.streamID)
+      let streamID = http2StreamID.map { streamID in
+        return String(Int(streamID))
```

Comment:
More nit, the return shouldn't be needed right? Entirely personal opinion, this would look really neat on one line. (I already approved, you can ignore).

Reply:
I tend to prefer explicit returns so I'l leave as is.